Enhancements and Fixes

Jan21493 opened the pull request:
Enhancements and Fixes
- Added columns for file size and file date
- Fixed calcuation for modification date when file list is retrieved from MS via FTP
  - handle modification date of Feb 29th (leap years) correctly
  - translate modification time from UTC to local time
- Added red background and tooltip, if file is not using UUID plus .yyyyMM as filename
- Added tooltips in header row
- Added button to delete file on MS and local file system
- Added button to convert a file (not ready yet)
- Stability improvements

Jan21493 committed:
Fixed two problems

Jan21493 committed:
Update .gitignore

Jan21493 committed:
Enhancements and Fixes
- Added column for file size and file date
- Fixed handling with timestamps in FTP transfers (UTC), adjustment of year
- Added marker, if file is not using UUID plus .yyyyMM as filename
- Added button to delete file on MS and local file system
- Added button to konvert a file (not ready yet)

Jan21493 committed:
- Fixed calcuation for modification date when file list is retrieved from MS via FTP
  - handle Feb 29 correctly
  - translate from UTC to local time
- Added tooltip for potentially malformed statistics filename
- Stability improvements

Jan21493 committed:
Minor Fixes

mr-manuel commented on MiniserverForm.Designer.cs:
```diff
@@ -351,28 +423,30 @@ private void InitializeComponent()
+            this.AutoScaleDimensions = new System.Drawing.SizeF(12F, 25F);
             this.AutoScaleMode = System.Windows.Forms.AutoScaleMode.Font;
-            this.ClientSize = new System.Drawing.Size(969, 561);
+            this.ClientSize = new System.Drawing.Size(2410, 1079);
```
Please do not increase the window size that much. Use the smallest size possible as start. There are a lot of users with 1.366 x 768 pixel screen size.

mr-manuel commented on MiniserverForm.Designer.cs:
```diff
@@ -393,7 +467,8 @@ private void InitializeComponent()
             this.Icon = ((System.Drawing.Icon)(resources.GetObject("$this.Icon")));
-            this.MinimumSize = new System.Drawing.Size(750, 270);
+            this.Margin = new System.Windows.Forms.Padding(6);
+            this.MinimumSize = new System.Drawing.Size(1474, 454);
```
This will make the program unusable on smaller screens like 1024 x 768 pixel.

Jan21493 replied:
I just had a look, but the reduced size might be not enough. I'm using a virtual machine on my Mac with retina display. The resolutions are quite high on such a system. I'll have a look on the issue tomorrow.

mr-manuel commented:
Thank you for contributing and improving this program!

> Added columns for file size and file date

Which info is displayed here? From remote or local filesystem? Could be confusing.

> Added button to convert a file (not ready yet)

What should this exactly do in future?

Jan21493 commented:
Hi Manuel,

here are my comments and questions to your questions:

> > Added columns for file size and file date
> 
> Which info is displayed here? From remote or local filesystem? Could be confusing.

Good point. It's always the local FS if the file is available on it. Reason: to avoid fetching of the directory listing via FTP after each 'edit'. At least the timestamp will be modified by 'edit'. Therefore I've changed the status message to "Older/Newer/Smaller/Larger on MS".
Proposed solutions: 
a) I add a tooltip with detailed information in each row for 'size' column with details (like MS: xx, FS: yy)
b) I modify the tooltip in the header row for the 'size' column with explanation like above.
Would one of the options solve the issue?
 
> > Added button to convert a file (not ready yet)
> 
> What should this exactly do in future?

I plan to migrate all my old statistics from the meter function block to the new format. It looks that this is possible. There is a user in the forum that has done that based on Excel and Script. (I can't remember the exact post, but I have referred to it in a recent post).

> Please do not increase the window size that much. Use the smallest size possible as start. 
> There are a lot of users with 1.366 x 768 pixel screen size.

The reason that I have increased the size via GUI was that the output looked ugly (size of each column), so I had to increase the size of the program after each start. I was not aware that this would not only change the 'default' size of the window, but also (or only) change the minimum size.

Proposed solution: 
- I set the size to the old values, but add a 'save window size' after each resize of the main window. These values are stored in the registry and used on startup if found. Would that solve the issue?

mr-manuel commented:
1. A simple tooltip with a short explanation should be enough

2. Thats a cool idea. In this case you have to verify, if the loaded statistic supports that and is the type you need as source.

3. In my opinion it would be nicer to save it to a *.ini file, so the Registry don't get cluttered, but don't add a button to save windows size. Just save it, after it changed. With this chance, you can also add, if the user wants to save the connection details.

That are great changes!

Jan21493 committed:
Enhancements and fixes
- Size of main window is reduced to be compatible for most users
- At start the window size in main window is resotred to previous settings
- Enhancements in tool tip messages to explain columns

Jan21493 commented:
Hi Manuel,
I've enhanced the tooltip messages in the header row to explain each column, especially size and modificaition date/time. The default windows size is smaller now and remembered. Stackoverflow was my friend ;-), see https://stackoverflow.com/questions/1873658/net-windows-forms-remember-windows-size-and-location
It's neither registry nor ini file, but xml file in userdata folder as recommended by Microsoft.

Please have a look.

Jan21493 committed:
Window size of main window is back to original size, so that the program will work with a screen size of 1024x768

Jan21493 committed:
Window size settings are maintained after upgrades of version number

Jan21493 commented:
Hi Manuel,
I've adjusted all sizes back to the original values. I believe that VisualStudio has adjusted (doubled) all sizes when I used the GUI with my large retina display. Anyway, this should be solved now.

After that I did a lot of tests and have increased the assembly version. Then I realized that the window size was back to default after the upgrade, so I added a few more lines of code. Overall, I believe that all topics have been fixed now.

mr-manuel commented:
Nice, thank you! I will test it this weekend and then merge the PR.
